src/osrm/osrm.cpp

```diff
@@ -15,7 +16,18 @@ namespace osrm
 
 // Pimpl idiom
 
-OSRM::OSRM(engine::EngineConfig &config) : engine_(std::make_unique<engine::Engine>(config)) {}
+OSRM::OSRM(engine::EngineConfig &config)
+{
+    if (engine::Engine<engine::algorithm::CoreCH>::CheckCompability(config))
+    {
+        engine_ = std::make_unique<engine::Engine<engine::algorithm::CoreCH>>(config);
+    }
```

Comment:
Added a log line in `Engine` that print shared vs. internal memory + algorithm used.